### PR TITLE
Fix slow response while undoing multiple selected entities changes

### PR DESCRIPTION
**Pull request by chiyenteng**
Bug fix for #3016.

When the user undoes changes for a larger group of selected entities (20+) the Editor noticeably hitches. This increases with the number of entities. At 40+ the hitch is multiple seconds and can cause the Editor to stop responding.

Root cause of this issue:

- Current system will destroy and recreate all these entities.
- While an entity destroyed, **Entity::Reset** is called and **OnEntityDestroyed** notification was sent.
- **EntityPropertyEditor::UpdateContents** is called every time when a selected entity gets destroyed, so the property editor will repeatedly refresh 20+ times, causing the very slow response.

Solution:
Replace **UpdateContents** with **QueuePropertyRefresh** so that we only refresh property editor once at next tick for multiple **UpdateContents** calls.

Before this fix undo 32 entities takes 60+ seconds. After this fix it only takes less than one second.

Signed-off-by: chiyenteng <82238204+chiyenteng@users.noreply.github.com>

### CODE REVIEW

**Review by AMZN-koppersr — APPROVED** [review]
If the history can be found, maybe see if was a reason to use `UpdateContents();` in the first place. The comment seems to suggest that an immediate refresh was needed for some reason.

**Review by NicholasVanSickle — APPROVED** [review]
Looks good to me - if we can find out why the immediate refresh happens, that'd be helpful, otherwise I suggest we just get a testing pass on the books for this.

**Comment by hultonha** [comment]
Great description and this sounds like a huge win, but definitely worth running some more tests to validate the change. As @AMZN-koppersr mentions this could well have been changed before to fix another subtle bug (which may no longer be relevant but always good to check). Perhaps speak with @jckand-amzn briefly if they have any ideas about edge cases and also would be worth checking with @AMZN-daimini and @cgalvan if they can think of any other things to test (or maybe even write some automation/integration tests to cover! 🙂

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/EntityPropertyEditor.cpp [hunk 1]
```diff
@@ -797,7 +797,7 @@ namespace AzToolsFramework
     {
         if (IsEntitySelected(entityId))
         {
-            UpdateContents(); // immediately refresh
+            QueuePropertyRefresh();
```
Comment (cgalvan) [inline Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/EntityPropertyEditor.cpp:800]: I went back in time and found this was added back in 2017 in order to fix an issue where pinned inspectors still showed the entity after a slice was deleted. So I think it is safe to do this. The `QueuePropertyRefresh` method already immediately disables actions/edits in the component editors and then queues the call to `UpdateContents`.

As for an automated test, it might even be better to create a benchmark for this workflow so we can make sure you can undo a lot of these changes and verifying it happens within a certain threshold.

**Review by cgalvan — APPROVED** [review]

**Review by chiyenteng — COMMENTED** [review]
Created an internal ticket LYN-11034 for automated tests/benchmarks for this workflow.

**Comment by chiyenteng** [comment]
> Looks good to me - if we can find out why the immediate refresh happens, that'd be helpful, otherwise I suggest we just get a testing pass on the books for this.

@dm-amzn has helped get a testing pass.

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/EntityPropertyEditor.cpp (hunk 1 above)
Comment (hultonha) [reply Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/EntityPropertyEditor.cpp:800]: Thanks very much for doing the detective work @cgalvan! Definitely be something to keep an eye on and a test+benchmark are worth adding. @chiyenteng could we add a follow-up GHI for this?

Comment (hultonha) [reply Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/EntityPropertyEditor.cpp:800]: Oh sorry I see you did this below 😅 Thanks!
